src/celeritas/em/interactor/MuBetheBlochInteractor.hh

```diff
@@ -0,0 +1,257 @@
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/data/StackAllocator.hh"
+#include "corecel/math/Algorithms.hh"
+#include "corecel/math/ArrayUtils.hh"
+#include "celeritas/Constants.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/em/data/MuBetheBlochData.hh"
+#include "celeritas/phys/CutoffView.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/phys/Secondary.hh"
+#include "celeritas/random/distribution/BernoulliDistribution.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+#include "detail/PhysicsConstants.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform the discrete part of the muon ionization process.
+ *
+ * This simulates the production of delta rays by incident mu- and mu+ with
+ * energies above 200 keV.
+ *
+ * \note This performs the same sampling routine as in Geant4's
+ * G4MuBetheBlochModel.
+ */
+class MuBetheBlochInteractor
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Mass = units::MevMass;
+    using Energy = units::MevEnergy;
+    using Momentum = units::MevMomentum;
+    //!@}
+
+  public:
+    //! Construct with shared and state data
+    inline CELER_FUNCTION
+    MuBetheBlochInteractor(MuBetheBlochData const& shared,
+                           ParticleTrackView const& particle,
+                           CutoffView const& cutoffs,
+                           Real3 const& inc_direction,
+                           StackAllocator<Secondary>& allocate);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Shared constant physics properties
+    MuBetheBlochData const& shared_;
+    // Allocate space for the secondary particle
+    StackAllocator<Secondary>& allocate_;
+    // Incident direction
+    Real3 const& inc_direction_;
+    // Incident particle energy [MeV]
+    real_type inc_energy_;
+    // Incident particle momentum [MeV / c]
+    real_type inc_momentum_;
+    // Square of fractional speed of light
+    real_type beta_sq_;
+    // Muon mass
+    real_type mass_;
+    // Electron mass
+    real_type electron_mass_;
+    // Total energy of the incident particle [MeV]
+    real_type total_energy_;
+    // Secondary electron cutoff for current material [MeV]
+    real_type electron_cutoff_;
+    // Maximum energy of the secondary electron [MeV]
+    real_type max_secondary_energy_;
+    // Whether to apply the radiative correction
+    bool use_rad_corr_;
+    // Envelope distribution for rejection sampling
+    real_type envelope_dist_;
+
+    //// CONSTANTS ////
+
+    //! Incident energy above which the radiative correction is applied [MeV]
+    static CELER_CONSTEXPR_FUNCTION Energy inc_energy_limit()
+    {
+        return Energy{250};  //!< 250 MeV
```

Comment:
```suggestion
        return units::MevEnergy{250};
```
since it's a hardcoded constant rather than an "internal representation of energy" (which is what the `Energy` typedef signifies)